Chi-square Test For Population Variance

$$\chi^2 = \frac{(n - 1)S^2}{\sigma^2}$$

where:

+ n is the sample size 

+ $S^2$ is the sample variance

+ $\sigma^2$ is the population variance

Exericse 1: An analysist is interested in the variance in time required to observe some medical effects of a newly developed drug at a clinical trial center. The drug developer will readjust the ingredients of the medicine if there is sufficient evidence (at $\alpha$ = 0.01) that the population variance exceeds 20 hours. A sample of 31 trails exhibits a sample variance of 31.2 hours. What do the results of this analysis suggest about the drug ingredient readjustment?

$$H_o: \sigma^2 \le 20$$

$$H_1: \sigma^2 > 20$$

In [5]:
n = 31
df = n - 1 # degrees of freedom
pop_var = 20
sample_var = 31.2
alpha = 0.01

In [12]:
# Compute test statistics chi_square_score
chi_square_score = ((n - 1)*sample_var)/pop_var
print("Test statistic Chi-square score =", chi_square_score)

# Compute critical value of right tailed test
from scipy.stats import chi2
critical_value = chi2.ppf(1-alpha, df)
print("Critical value =", critical_value)

p_value = 1 - chi2.cdf(chi_square_score, df)
print("p_value = ",p_value)

Test statistic Chi-square score = 46.8
Critical value = 50.89218131151707
p_value =  0.026025000893334038


Since the test statistic Chi-square score < critical value (46.8 < 50.892)

We do not reject $H_o$

Conclusion: The developer does not need to readjust the drug ingredient since the population does not exceed 20 hours.